tufaceous-lib/src/repository.rs

```diff
@@ -276,15 +278,13 @@ impl OmicronRepoEditor {
             artifact.target = filename.clone();
         } else {
```

Comment:
In the `if` part of this branch, we're accepting (and replacing) an artifact if we're called with the same name+version+kind. I'm nervous about that - it certainly seems wrong if we're building from a manifest (since we shouldn't have duplicates like that, and the existence of a such a dup probably means we've made a mistake), but maybe it's necessary in general?

Reply:
I think this should probably error, yeah. You're right that the existence of a duplicate name-version-kind triple means we've more likely made a mistake.

Reply:
Changed to an error in [7898a35](https://github.com/oxidecomputer/omicron/pull/3990/commits/7898a35605936d620130316ff520f18f200fef67)